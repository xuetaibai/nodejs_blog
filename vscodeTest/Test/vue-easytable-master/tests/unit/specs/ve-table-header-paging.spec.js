import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";

describe("veTable header paging", () => {
    const TABLE_DATA = [
        {
            name: "John",
            age: 25,
            weight: 66,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shanghai",
        },
        {
            name: "Dickerson",
            age: 20,
            weight: 70,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Beijing",
        },
        {
            name: "Larsen",
            age: 18,
            weight: 65,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
        },
        {
            name: "Geneva",
            age: 17,
            weight: 80,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
        },
        {
            name: "Jami",
            age: 26,
            weight: 72,
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
        },
    ];

    it("render single field paging", () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const wrapper = mount({
            render() {
                return (
                    <veTable
                        sortOption={this.sortOption}
                        columns={this.columns}
                        tableData={this.tableData}
                        rowKeyFieldName="rowkey"
                    />
                );
            },
            data() {
                return {
                    columns: columns,
                    tableData: TABLE_DATA,
                    sortOption: {
                        sortChange: (params) => {},
                    },
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render multiple field paging", () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const wrapper = mount({
            render() {
                return (
                    <veTable
                        sortOption={this.sortOption}
                        columns={this.columns}
                        tableData={this.tableData}
                        rowKeyFieldName="rowkey"
                    />
                );
            },
            data() {
                return {
                    columns: columns,
                    tableData: TABLE_DATA,
                    sortOption: {
                        multipleSort: true,
                        sortChange: (params) => {},
                    },
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("single field paging", async () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const mockFn = jest.fn();
        const callBackData = {
            age: "asc",
            weight: "",
        };

        const wrapper = mount(veTable, {
            propsData: {
                columns: columns,
                tableData: TABLE_DATA,
                sortOption: {
                    sortChange: (params) => {
                        mockFn(params);
                    },
                },
                rowKeyFieldName: "rowkey",
            },
        });

        const thEls = wrapper.findAll(".ve-table-header-tr th");
        expect(thEls.at(1).find(".ve-table-sort").exists()).toBe(true);

        expect(
            thEls
                .at(2)
                .findAll(".ve-table-sort .ve-table-sort-icon")
                .at(0)
                .find(".active")
                .exists(),
        ).toBe(true);

        thEls.at(1).find(".ve-table-sort").trigger("click");

        await later();

        expect(mockFn).toBeCalled();
        expect(mockFn).toHaveBeenCalledWith(callBackData);

        expect(
            thEls
                .at(1)
                .findAll(".ve-table-sort .ve-table-sort-icon")
                .at(0)
                .find(".active")
                .exists(),
        ).toBe(true);
    });

    it("multiple field paging", async () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const mockFn = jest.fn();
        const callBackData = {
            age: "asc",
            weight: "asc",
        };

        const wrapper = mount(veTable, {
            propsData: {
                columns: columns,
                tableData: TABLE_DATA,
                sortOption: {
                    multipleSort: true,
                    sortChange: (params) => {
                        mockFn(params);
                    },
                },
                rowKeyFieldName: "rowkey",
            },
        });

        const thEls = wrapper.findAll(".ve-table-header-tr th");
        expect(thEls.at(1).find(".ve-table-sort").exists()).toBe(true);

        expect(
            thEls
                .at(2)
                .findAll(".ve-table-sort .ve-table-sort-icon")
                .at(0)
                .find(".active")
                .exists(),
        ).toBe(true);

        thEls.at(1).find(".ve-table-sort").trigger("click");

        await later();

        expect(mockFn).toBeCalled();
        expect(mockFn).toHaveBeenCalledWith(callBackData);

        expect(
            wrapper.findAll(".ve-table-sort .ve-table-sort-icon.active").length,
        ).toBe(2);
    });

    it("sort always", async () => {
        const columns = [
            {
                field: "name",
                key: "a",
                title: "Name",
                align: "left",
            },
            {
                field: "age",
                key: "b",
                title: "Age",
                align: "center",
                sortBy: "",
            },
            {
                field: "weight",
                key: "c",
                title: "Weight(kg)",
                align: "center",
                sortBy: "asc",
            },
            {
                field: "hobby",
                key: "d",
                title: "Hobby",
                align: "center",
            },
            { field: "address", key: "e", title: "Address", align: "left" },
        ];

        const mockFn = jest.fn();

        const wrapper = mount(veTable, {
            propsData: {
                columns: columns,
                tableData: TABLE_DATA,
                sortOption: {
                    sortAlways: true,
                    sortChange: (params) => {
                        mockFn(params);
                    },
                },
                rowKeyFieldName: "rowkey",
            },
        });

        const thEls = wrapper.findAll(".ve-table-header-tr th");

        thEls.at(1).find(".ve-table-sort").trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalledWith({
            age: "asc",
            weight: "",
        });

        thEls.at(1).find(".ve-table-sort").trigger("click");

        await later();

        expect(mockFn).toHaveBeenCalledWith({
            age: "desc",
            weight: "",
        });
    });
});
