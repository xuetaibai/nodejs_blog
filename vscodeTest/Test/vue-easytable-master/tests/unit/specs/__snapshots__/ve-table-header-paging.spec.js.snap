// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable header paging render multiple field paging 1`] = `
<div class="ve-table">
  <div class="ve-table-container ve-table-border-around" style="height: 2px;">
    <table class="ve-table-content ve-table-border-x">
      <colgroup>
        <col>
        <col>
        <col>
        <col>
        <col>
      </colgroup>
      <thead class="ve-table-fixed-header ve-table-header">
        <tr class="ve-table-header-tr">
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: left; top: 0px;">Name</th>
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: center; top: 0px;">Age<span class="ve-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow ve-table-sort-icon ve-table-sort-icon-top"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow ve-table-sort-icon ve-table-sort-icon-bottom"></i></span></th>
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: center; top: 0px;">Weight(kg)<span class="ve-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow ve-table-sort-icon ve-table-sort-icon-top active"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow ve-table-sort-icon ve-table-sort-icon-bottom"></i></span></th>
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
          <th rowspan="1" colspan="1" class="ve-table-header-th ve-table-last-column" style="text-align: left; top: 0px;">Address</th>
        </tr>
      </thead>
      <tbody class="ve-table-body ve-table-row-hover ve-table-row-highlight">
        <tr style="height: 0px;">
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">John</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">25</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">66</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Dickerson</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">20</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">70</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Larsen</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">18</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">65</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Geneva</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">17</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">80</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Jami</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">26</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">72</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
        </tr>
      </tbody>
      <tfoot class="ve-table-fixed-footer ve-table-footer"></tfoot>
    </table>
  </div>
</div>
`;

exports[`veTable header paging render single field paging 1`] = `
<div class="ve-table">
  <div class="ve-table-container ve-table-border-around" style="height: 2px;">
    <table class="ve-table-content ve-table-border-x">
      <colgroup>
        <col>
        <col>
        <col>
        <col>
        <col>
      </colgroup>
      <thead class="ve-table-fixed-header ve-table-header">
        <tr class="ve-table-header-tr">
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: left; top: 0px;">Name</th>
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: center; top: 0px;">Age<span class="ve-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow ve-table-sort-icon ve-table-sort-icon-top"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow ve-table-sort-icon ve-table-sort-icon-bottom"></i></span></th>
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: center; top: 0px;">Weight(kg)<span class="ve-table-sort"><i class="ve-icon iconfont-vet icon-vet-sort-top-arrow ve-table-sort-icon ve-table-sort-icon-top active"></i><i class="ve-icon iconfont-vet icon-vet-sort-bottom-arrow ve-table-sort-icon ve-table-sort-icon-bottom"></i></span></th>
          <th rowspan="1" colspan="1" class="ve-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
          <th rowspan="1" colspan="1" class="ve-table-header-th ve-table-last-column" style="text-align: left; top: 0px;">Address</th>
        </tr>
      </thead>
      <tbody class="ve-table-body ve-table-row-hover ve-table-row-highlight">
        <tr style="height: 0px;">
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
          <td style="padding: 0px; border: 0px; height: 0px;"></td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">John</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">25</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">66</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Dickerson</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">20</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">70</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Larsen</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">18</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">65</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Geneva</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">17</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">80</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
        </tr>
        <tr class="ve-table-body-tr">
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">Jami</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">26</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">72</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: center;">coding and coding repeat</td>
          <td rowspan="1" colspan="1" class="ve-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
        </tr>
      </tbody>
      <tfoot class="ve-table-fixed-footer ve-table-footer"></tfoot>
    </table>
  </div>
</div>
`;
